packages/flutter/test/widgets/editable_text_test.dart

```diff
@@ -0,0 +1,61 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets('Changing controller updates EditableText', (WidgetTester tester) async {
+    final GlobalKey<EditableTextState> editableTextKey = new GlobalKey<EditableTextState>();
+    final TextEditingController controller1 = new TextEditingController(text: 'Wibble');
+    final TextEditingController controller2 = new TextEditingController(text: 'Wobble');
+    TextEditingController currentController = controller1;
+    StateSetter setState;
+
+    Widget builder() {
+      return new StatefulBuilder(
+        builder: (BuildContext context, StateSetter setter) {
+          setState = setter;
+          return new Directionality(
+            textDirection: TextDirection.ltr,
+            child: new Center(
+              child: new Material(
+                child: new EditableText(
+                  key: editableTextKey,
+                  controller: currentController,
+                  focusNode: new FocusNode(),
+                  style: new Typography(platform: TargetPlatform.android).black.subhead,
+                  cursorColor: Colors.blue,
+                  selectionControls: materialTextSelectionControls,
+                  keyboardType: TextInputType.text,
+                  onChanged: (String value) {},
+                ),
+              ),
+            ),
+          );
+        },
+      );
+    }
+    await tester.pumpWidget(builder());
+    await tester.showKeyboard(find.byType(EditableText));
+
+    // Verify TextInput.setEditingState is fired with updated text when controller is replaced.
+    final List<MethodCall> log = <MethodCall>[];
+    SystemChannels.textInput.setMockMethodCallHandler((MethodCall methodCall) {
+      log.add(methodCall);
+    });
+    setState(() {
+      currentController = controller2;
+    });
+    await tester.pump();
+
+    final MethodCall setStateCall = log.firstWhere((MethodCall methodCall) {
+      return methodCall.method == 'TextInput.setEditingState';
+    });
```

Comment:
would it make sense to be more precise about what should be in the log? e.g. that it only has one entry, or whatnot? What if there were two calls, one saying Wobble and one saying Wibble?

Reply:
Done. Was worried about testing some assumptions we don't care about here but the result is under-testing, and if someone's going to tweak this code, we should probably force them to check these tests anyway :-)